pkg/installmanager/aws_credentials.go

```diff
@@ -0,0 +1,156 @@
+package installmanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	contributils "github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/awsclient"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+// AWSCredentials is a supported external process credential provider as detailed in
+// https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html
+type AWSCredentials struct {
+	output     io.WriteCloser
+	kubeClient client.Client
+
+	ServiceProviderSecretName      string
+	ServiceProviderSecretNamespace string
+
+	RoleARN    string
+	ExternalID string
+}
+
+// NewInstallManagerAWSCredentials is the entrypoint to load credentials for AWS SDK
+// using the service provider credentials. It supports the external process credential
+// provider as mentioned in https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html
+func NewInstallManagerAWSCredentials() *cobra.Command {
+	options := &AWSCredentials{}
+	cmd := &cobra.Command{
+		Use:   "aws-credentials",
+		Short: "Loads AWS credentials using the service provider credentials",
+		Long:  "This loads the AWS credentials using the service provider credentials and then returns them in format defined in https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(args); err != nil {
+				fmt.Fprint(os.Stderr, err.Error())
+				return
+			}
+
+			if err := options.Validate(); err != nil {
+				fmt.Fprint(os.Stderr, err.Error())
+				os.Exit(1)
+				return
+			}
+
+			var err error
+			options.kubeClient, err = contributils.GetClient()
+			if err != nil {
+				fmt.Fprint(os.Stderr, err.Error())
+				os.Exit(1)
+				return
+			}
+
+			if err := options.Run(); err != nil {
+				fmt.Fprint(os.Stderr, err.Error())
+				os.Exit(1)
+				return
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&options.ServiceProviderSecretNamespace, "namespace", "", "The namespace where the service provider secret is stored")
+	cmd.MarkFlagRequired("namespace")
+	flags.StringVar(&options.RoleARN, "role-arn", "", "The IAM role that should be assumed")
+	cmd.MarkFlagRequired("role-arn")
+	flags.StringVar(&options.ExternalID, "external-id", "", "External identifier required to assume the role specified.")
+
+	return cmd
+}
+
+// Validate the options
+func (options *AWSCredentials) Validate() error { return nil }
+
+// Complete the options using the args
+func (options *AWSCredentials) Complete(args []string) error {
+	options.output = os.Stdout
+	options.ServiceProviderSecretName = os.Getenv(constants.HiveAWSServiceProviderCredentialsSecretrefEnvVar)
```

Comment:
Do we have a good reason to split between CLI args and env vars for this? It seems odd to have --namespace for the Secret namespace, but the Secret name from an env var.

Reply:
the hive controllers get the servicer p secret from the environment and the namespace is constant. all the code in controller uses this way to setup aws client. 

in the install/uninstall pods, any hive code that needs aws client will probably look to controller code for consistency.
making sure the source of secret name comes from the same env makes things consistent.

the namespace is not something install/uninstall pods has access to and therefore needs specifying.

lastly we can configure the hive controllers such that we don't need the HiveAWSServiceProviderCredentialsSecretrefEnvVar to provide credentials, the environment is good enough.
in that case it's easier to say i don't have the secret env, so these pods shouldn't too assuming that there is environment setup to fulfill those credentials.